test/gossip.js

```diff
@@ -0,0 +1,102 @@
+'use strict'
+/* eslint-env mocha */
+
+const { expect } = require('chai')
+const sinon = require('sinon')
+
+const { GossipSubDhi } = require('../src/constants')
+const {
+  first,
+  createNode,
+  dialNode,
+  startNode,
+  stopNode
+} = require('./utils')
+
+describe('gossip', () => {
+  let nodes = Array.from({ length: GossipSubDhi + 2 }) // enough nodes to trigger high threshold
+
+  beforeEach(async () => {
+    for (let i = 0; i < nodes.length; i++) {
+      nodes[i] = await createNode('/ip4/127.0.0.1/tcp/0')
+      await startNode(nodes[i])
+      await startNode(nodes[i].gs)
+    }
+  })
+  afterEach(async function () {
+    this.timeout(6000)
+    await Promise.all(nodes.map((n) => stopNode(n.gs)))
+    await Promise.all(nodes.map((n) => stopNode(n)))
+  })
+
+  it('should send gossip to non-mesh peers in topic', async function () {
+    this.timeout(0)
```

Comment:
Why this?

Reply:
It took longer than 2 seconds, ~5 seconds. I can try to dial it in so it is at least bounded.